### PR TITLE
hugs: fix build on darwin

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Fix build on darwin. Had to pick some commits from a [repository](https://github.com/FranklinChen/hugs98-plus-Sep2006) that was working on the same issue.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/hugs/default.nix [hunk 1]
```diff
@@ -11,18 +11,42 @@ stdenv.mkDerivation {
...
+        url = "https://github.com/FranklinChen/hugs98-plus-Sep2006/commit/5c4a0d18f2a941839947fc0c2126374e95164802.patch";
+        sha256 = "190siizla95famlp80rsqhmyqy5al9mz6qg5zix04hqxkv3c6wr2";
+      })
+      (fetchpatch {
+        url = "https://github.com/FranklinChen/hugs98-plus-Sep2006/commit/4f323b9ab1fd6c53fe48d72aad4e7fba2ce8850c.patch";
+        sha256 = "1azzylllbbwvfrdfljk9wvkj8gcqxj3nmxya65xbffkrw1gp3kh2";
       })
     ];
 
+  buildInputs = [ gcc ];
```
Comment (symphorien) [inline pkgs/development/interpreters/hugs/default.nix:40]: you probably meant to use gccStdenv ?

Comment (siraben) [reply pkgs/development/interpreters/hugs/default.nix:40]: Oops, yes.

**Comment by siraben** [comment]
@symphorien changing `gcc` to `gccStdenv` breaks the binary at runtime, giving

```
./result/bin/hugs 
__   __ __  __  ____   ___      _________________________________________
||   || ||  || ||  || ||__      Hugs 98: Based on the Haskell 98 standard
||___|| ||__|| ||__||  __||     Copyright (c) 1994-2005
||---||         ___||           World Wide Web: http://haskell.org/hugs
||   ||                         Bugs: http://hackage.haskell.org/trac/hugs
||   || Version: September 2006 _________________________________________

Haskell 98 mode: Restart with command line option -98 to enable extensions

Prelude not found on current path: ".:{Home}/lib/hugs/packages/*:/usr/local/lib/hugs/packages/*:{Hugs}/packages/*"

FATAL ERROR: Unable to load prelude
```

**Comment by symphorien** [comment]
I have no idea why that would cause it, sorry.

**Comment by siraben** [comment]
@joachifm

**Comment by siraben** [comment]
@jtojnar could you help take a look as well?

**Comment by jtojnar** [comment]
gccStdenv would be used in place of regular stdenv, not in buildInputs. 
But what is the issue with Clang exactly?

**Comment by siraben** [comment]
@jtojnar here is the issue when I use clang to build
```
$ ./result/bin/hugs
__   __ __  __  ____   ___      _________________________________________
||   || ||  || ||  || ||__      Hugs 98: Based on the Haskell 98 standard
||___|| ||__|| ||__||  __||     Copyright (c) 1994-2005
||---||         ___||           World Wide Web: http://haskell.org/hugs
||   ||                         Bugs: http://hackage.haskell.org/trac/hugs
||   || Version: September 2006 _________________________________________

Haskell 98 mode: Restart with command line option -98 to enable extensions

Prelude not found on current path: ".:{Home}/lib/hugs/packages/*:/usr/local/lib/hugs/packages/*:{Hugs}/packages/*"

FATAL ERROR: Unable to load prelude
```

File: pkgs/development/interpreters/hugs/default.nix [hunk 2]
```diff
@@ -11,18 +11,42 @@ stdenv.mkDerivation {
 
   patches =
     [ (fetchurl {
-        url = "https://aur.archlinux.org/cgit/aur.git/plain/hsbase_inline.patch?h=hugs";
-        name = "hsbase_inline.patch";
-        sha256 = "1h0sp16d17hlm6gj7zdbgwrjwi2l4q02m8p0wd60dp4gn9i9js0v";
+      url = "https://aur.archlinux.org/cgit/aur.git/plain/hsbase_inline.patch?h=hugs";
+      name = "hsbase_inline.patch";
+      sha256 = "1h0sp16d17hlm6gj7zdbgwrjwi2l4q02m8p0wd60dp4gn9i9js0v";
+    }) ] ++ lib.optionals stdenv.isDarwin [
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/hugs/default.nix:17]: Please always apply all patches.

File: pkgs/development/interpreters/hugs/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/development/interpreters/hugs/default.nix:40]: nativeBuildInputs.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 100216` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>hugs</li>
</ul>
</details>

**Comment by siraben** [comment]
I'll apply those changes.

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.

**Pull request closed without merging**